src/ray/util/event.cc

```diff
@@ -323,6 +381,40 @@ void RayEvent::SendMessage(const std::string &message) {
   }
 }
 
+///
+/// RayExportEvent
+///
+RayExportEvent::~RayExportEvent() {}
+
+void RayExportEvent::SendEvent() {
+  if (EventManager::Instance().IsEmpty()) {
+    return;
+  }
+
+  static const int kEventIDSize = 18;
+  std::string event_id;
+  std::string event_id_buffer = std::string(kEventIDSize, ' ');
+  FillRandom(&event_id_buffer);
+  event_id = StringToHex(event_id_buffer);
+
+  rpc::ExportEvent export_event;
+  export_event.set_event_id(event_id);
+  export_event.set_source_type(source_type_);
+  export_event.set_timestamp(current_sys_time_s());
+
+  std::string event_data_type_name = event_data_ptr_->GetTypeName();
```

Comment:
This seems pretty fragile. What about we do this instead?

1. We create a new message EventData
```
msg EventData {
    oneof {
       ExportTaskEventData ...
    }
}
```

To the class, we require to pass EventData instead

Reply:
Good idea. I did something similar using `std::variant` instead so the caller doesn't have to construct an additional proto message. This allows us to check the type with `std::get_if` so there is no string comparison